src/jit/compiler.cpp

```diff
@@ -4598,6 +4598,10 @@ void Compiler::compCompile(void** methodCodePtr, ULONG* methodCodeSize, JitFlags
         codeGen->regSet.rsMaskResvd |= RBM_OPT_RSVD;
         assert(REG_OPT_RSVD != REG_FP);
     }
+    // compRsvdRegCheck() has read out the FramePointerUsed property, but doLinearScan()
+    // tries to overwrite it later. This violates the PhasedVar rule and triggers an assertion.
+    // TODO: What is the proper way to handle this situation?
+    codeGen->resetFramePointerUsedWritePhase();
```

Comment:
This probably requires some further analysis, but I don't think that the `rsMaskResvd` should be used in the non-LEGACY_BACKEND case, since these reserved registers should be handled explicitly by the register allocator. I see, however, that there are several references to this in the `_TARGET_ARM_` case.